resources/definitions/kingroon_base.def.json

```diff
@@ -0,0 +1,271 @@
+{
+    "name": "Kingroon Base Printer",
+    "version": 2,
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": false,
+        "author": "notamu",
+        "manufacturer": "Kingroon",
+        "file_formats": "text/x-gcode",
+        "first_start_actions": ["MachineSettingsAction"],
+
+        "machine_extruder_trains": {
+            "0": "kingroon_base_extruder_0"
+        },
+
+        "has_materials": true,
+        "has_variants": true,
+        "has_machine_quality": true,
+        "variants_name": "Nozzle Size",
+
+        "preferred_variant_name": "0.4mm Nozzle",
+        "preferred_quality_type": "standard",
+        "preferred_material": "generic_pla",
+        "exclude_materials": [
```

Comment:
You define a material diameter of 1.75mm. However you exclude a lot of materials here that are not compatible with that. They don't need to be excluded since Cura already filters them out.

Reply:
I'm having troubles finding out what materials I should and shouldn't exclude. Is there any suggestion about how to properly do this?

Reply:
I think I have resolved this: I cleared the list and added one by one the materials that were listed by Cura (the ones without qualities, of course). I'm waiting for your review 🙂

Reply:
The "expected" behaviour is that you'd only exclude materials that you are certain won't work. Like, exclude polycarbonate materials because they require a printing temperature of 280C which is hotter than what the printer can reach. It's an open filament system after all, so people should be able to provide new profiles for the printer. We shouldn't limit people to use a combination of profiles if they really want to use their favourite filament in the Kingroon printers.
So in theory, you'd test those profiles one by one, and exclude them only if you tested that they don't work.

Understandably resources are limited and some guesswork is needed, but that's the workflow we've envisioned.